src/core/public/chrome/recently_accessed/recently_accessed_service.ts

```diff
@@ -59,14 +63,25 @@ export class RecentlyAccessedService {
 
     return {
       /** Adds a new item to the history. */
-      add: (link: string, label: string, id: string) => {
+      add: (
+        link: string,
+        label: string,
+        id: string,
+        extraProps?: ChromeRecentlyAccessedHistoryItem['extraProps']
+      ) => {
         const currentWorkspaceId = workspaces.currentWorkspaceId$.getValue();
 
         history.add({
           link,
           label,
           id,
           ...(currentWorkspaceId && { workspaceId: currentWorkspaceId }),
+          ...(extraProps?.type && {
+            extraProps: {
+              viewedAt: Date.now(),
```

Comment:
We could make `viewedAt` a property that generally for all recent accessed items, not only the one that passed `extraProps?.type`, how does that sound? And what about rename `viewedAt` -> `lastAccessedTime`?

Reply:
Sure, will refactor the code to reduce confusion.